testsuite/testcases/src/validator_join_leave_test.rs

```diff
@@ -0,0 +1,218 @@
+    }
+
+    fn test(&self, swarm: &mut dyn Swarm, duration: Duration) -> Result<()> {
+        // Verify we have at least 7 validators (i.e., 3f+1, where f is 2)
+        // so we can lose 2 validators but still make progress.
+        let all_validators = swarm.validators().map(|v| v.peer_id()).collect::<Vec<_>>();
+        let num_validators = all_validators.len();
+        if num_validators < 7 {
+            return Err(anyhow::format_err!(
+                "ValidatorSet leaving and rejoining test require at least 7 validators! Given: {:?}.",
+                num_validators
+            ));
+        }
+
+        let faucet_endpoint: reqwest::Url = "http://localhost:8081".parse().unwrap();
+        // Connect the operator tool to the node's JSON RPC API
+        let rest_client = swarm.validators().next().unwrap().rest_client();
+        let transaction_factory = swarm.chain_info().transaction_factory();
+        let runtime = Runtime::new().unwrap();
+
+        let mut cli = runtime.block_on(async {
+            CliTestFramework::new(
+                swarm.validators().next().unwrap().rest_api_endpoint(),
+                faucet_endpoint,
+                /*num_cli_accounts=*/ 0,
+            )
+            .await
+        });
+
+        let mut public_info = swarm.chain_info().into_aptos_public_info();
+
+        let mut validator_cli_indices = Vec::new();
+
+        let starting_seed_in_decimal = i64::from_str_radix(FORGE_KEY_SEED, 16)?;
+
+        for i in 0..num_validators {
+            // Initialize keyGen to get validator private keys. We uses the same seed in the test
+            // driver as in the genesis script so that the validator keys are deterministic.
+            let mut seed_slice = [0u8; 32];
+            let seed_in_decimal = starting_seed_in_decimal + (i as i64);
+            let seed_in_hex_string = format!("{seed_in_decimal:0>64x}");
+
+            hex::decode_to_slice(seed_in_hex_string, &mut seed_slice)?;
+
+            let mut keygen = KeyGen::from_seed(seed_slice);
+
+            let (validator_cli_index, _keys) = runtime.block_on(async {
+                let (validator_cli_index, keys) =
+                    init_validator_account(&mut cli, &mut keygen).await;
+
+                let auth_key = AuthenticationKey::ed25519(&keys.account_private_key.public_key());
+                let validator_account_address = AccountAddress::new(*auth_key.derived_address());
+
+                public_info
+                    .mint(validator_account_address, DEFAULT_FUNDED_COINS)
+                    .await
+                    .unwrap();
+                cli.assert_account_balance_now(validator_cli_index, DEFAULT_FUNDED_COINS)
+                    .await;
+                (validator_cli_index, keys)
+            });
+            validator_cli_indices.push(validator_cli_index);
+
+            assert_eq!(
+                runtime.block_on(async { get_validator_state(&cli, validator_cli_index).await }),
```

Comment:
you don't need `async { }` block to wrap the async function.  just `runtime.block_on(get_validator_stat(..` should do?

Reply:
Removing the `async` block leads to this error.
```
Forge output:    Compiling testcases v0.0.0 (/Users/zcc/aptos-core/testsuite/testcases)
error[E0728]: `await` is only allowed inside `async` functions and blocks
   --> testsuite/testcases/src/validator_join_leave_test.rs:98:80
    |
36  | /     fn test(&self, swarm: &mut dyn Swarm, duration: Duration) -> Result<()> {
37  | |         // Verify we have at least 7 validators (i.e., 3f+1, where f is 2)
38  | |         // so we can lose 2 validators but still make progress.
39  | |         let all_validators = swarm.validators().map(|v| v.peer_id()).collect::<Vec<_>>();
...   |
98  | |                 runtime.block_on(get_validator_state(&cli, validator_cli_index).await ),
    | |                                                                                ^^^^^^ only allowed inside `async` functions and blocks
...   |
164 | |         Ok(())
165 | |     }
    | |_____- this is not `async`

error[E0277]: `ValidatorState` is not a future
   --> testsuite/testcases/src/validator_join_leave_test.rs:98:34
    |
98  |                 runtime.block_on(get_validator_state(&cli, validator_cli_index).await ),
    |                         -------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `ValidatorState` is not a future
    |                         |
    |                         required by a bound introduced by this call
    |
    = help: the trait `futures::Future` is not implemented for `ValidatorState`
    = note: ValidatorState must be a future or must implement `IntoFuture` to be awaited
note: required by a bound in `tokio::runtime::Runtime::block_on`
   --> /Users/zcc/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.0/src/runtime/mod.rs:483:28
    |
483 |         pub fn block_on<F: Future>(&self, future: F) -> F::Output {
    |                            ^^^^^^ required by this bound in `tokio::runtime::Runtime::block_on`

Some errors have detailed explanations: E0277, E0728.
For more information about an error, try `rustc --explain E0277`.
error: could not compile `testcases` due to 2 previous errors
```

Reply:
Oh, I think you don't need `.await`.  `runtime.block_on(get_validator_state(&cli, validator_cli_index))` should do.